src/v/raft/group_manager.cc

```diff
@@ -79,6 +79,16 @@ ss::future<> group_manager::remove(ss::lw_shared_ptr<raft::consensus> c) {
       });
 }
 
+ss::future<> group_manager::shutdown(ss::lw_shared_ptr<raft::consensus> c) {
```

Comment:
I think we should think about this decision to move *all* leaders/etc. 

Is there a design document that describes this movement somewhere? 
My fear here is that it will cause a cluster-wide unavailability ***iff** we move the leaders. 

My hunch is that if we *only*** move the replicas (all non leader partitions) we might get a good enough balance without any meaningful disruption. 

I don’t see a way in this PR where we prevent movement of the active leaders. Maybe this policy is enforced at a higher level via a policy object. 

@dotnwat how were you thinking about policies for the client side rebalancing tool `rpk data rebalance —with-leaders=true —rate=0.1` or smith like that to specify the rate of change in a cluster?

Reply:
This is only low level API (whole partition movement mechanism) it is generic and support all the use cases. The rebalance logic will take care about not moving active leaders. Raft doesn't prevent changing membership when node is a leader, it is handled gracefully.

Reply:
this is important, but off topic